Closes #2227 re-add linked image in az_person az_row view mode.

trackleft opened the pull request:
## Description
Adding new link field group to az_row display mode for az_person content type with the following settings.
- target: custom_uri
- custom_uri: '[node:az-canonical-url]'

## Related issues
Closes #2227

## How to test

go to /people-list
Click on images of ducks to go to the profile page.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

bberndt-uaz commented on modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_row.yml:
```diff
@@ -141,6 +141,21 @@ third_party_settings:
+      weight: 23
+      format_type: link
+      format_settings:
+        classes: ''
```
Before converting AZ Person to use the Drupal UI, we added the `az-person-row-img` class to the link containing the image and then used [a rule in az_person.css](https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_person/css/az_person.css) to hide the contextual edit button. The pencil icon button for media doesn't work in this case since clicking it simply takes you to the full node.

If we want to keep that button hidden like it was before, we could add the `az-person-row-img` class to the field group (suggested below) or we could update the selector in az_person.css to replace `az-person-row-img` with `field-group-link`.

```suggestion
        classes: az-person-row-img
```

joshuasosa replied:
Could also maybe use `az-person-image-link` as used on the card display.

joshuasosa commented:
Doesn't really matter, but the title for the link group now has a few variations. On card, it's just labeled `Link`. In this PR for row, it gets labeled `Linked image`. For linked titles on both card and row, they're labeled `Title Link` and `Title link` respectively.

trackleft commented on modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_row.yml:
```diff
@@ -141,6 +141,21 @@ third_party_settings:
+    group_linked_image:
+      children:
+        - field_az_media_image
+      label: 'Linked image'
```
```suggestion
      label: 'Image link'
```

trackleft commented on modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_row.yml:
```diff
@@ -141,6 +141,21 @@ third_party_settings:
         attributes: ''
         effect: none
         speed: fast
+    group_linked_image:
```
```suggestion
    group_link:
```

trackleft commented on modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_row.yml:
```diff
@@ -141,6 +141,21 @@ third_party_settings:
+      weight: 23
+      format_type: link
+      format_settings:
+        classes: az-person-row-img
```
```suggestion
        classes: 'az-person-row-img az-person-image-link'
```

trackleft commented on modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_card.yml:
```diff
@@ -70,7 +70,7 @@ third_party_settings:
       children:
         - field_az_media_image
-      label: Link
+      label: Image link
```
```suggestion
      label: Image Link
```

trackleft commented on modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_row.yml:
```diff
@@ -141,6 +141,21 @@ third_party_settings:
+    group_link:
+      children:
+        - field_az_media_image
+      label: 'Image link'
```
```suggestion
      label: 'Image Link'
```

trackleft commented:
THanks All! I've applied the following changes

- Updated the az_person **card** display mode's Link field group label to Image Link to match the "Title Link" pattern. Did not update the field_group machine name, because it could cause weird update issues.
- Updated the az_person css file to use the correct base selector
- Updated the az_person **row** display mode's  Link field group label to Image Link to match the "Title Link" pattern.
- Updated the az_person **row** display mode's  Link field group machine name to match the card display mode
- Added the two suggested classes to the link field group for image
- Updated existing "link" field group labels to all use Title Case.

bberndt-uaz commented on modules/custom/az_person/css/az_person.css:
```diff
@@ -27,7 +27,7 @@
 }
 
-.az-person-grid .node--type-az-person.node--view-mode-az-row .az-person-row-img .contextual button {
+.az-person-row .node--type-az-person.node--view-mode-az-row .az-person-row-img .contextual button {
```
Could we remove `az-person-row-img` from the group classes and replace it with `az-person-image-link` here?

If we wanted to, we could even consolidate this rule with the one above and target `az-person-image-link` since we are only using it on these two displays. A selector such as `.contextual-region.node--type-az-person .az-person-image-link .contextual button` would work.

trackleft replied:
I think I would prefer a `.hide-contextual-links` class

trackleft commented on modules/custom/az_person/config/install/core.entity_view_display.node.az_person.az_row.yml:
```diff
@@ -141,6 +141,21 @@ third_party_settings:
+      weight: 23
+      format_type: link
+      format_settings:
+        classes: hide-contextual-links
```
Add this back because: https://github.com/az-digital/az_quickstart/blob/2.5.x/themes/custom/az_barrio/templates/content/node--az-person--az_row.html.twig#L87
```suggestion
        classes: 'hide-contextual-links az-person-row-img'
```

bberndt-uaz reviewed (approved):
Tested locally and confirmed that the new class is present and the behavior matches 2.5.

I noticed another issue that should not block this PR since it is present in 2.5 as well: when you click the contextual button for a person but don't follow the Edit or Delete links, the contextual button for the image for that person becomes visible since an inline style sets its display to `inline-block`, overriding the 'display: none' from the CSS styles.

The contextual button for the image begins with this HTML:

```html
<button class="trigger focusable visually-hidden" type="button" aria-pressed="false">
  Open configuration options</button>
```

When you click to open the contextual menu for the person node, a style attribute is added to the button for the image:

```html
<button class="trigger focusable visually-hidden" type="button" aria-pressed="false" 
  style="display: none;">Open configuration options</button>
```

Finally, when you click away from the opened menu so that it closes, the style attribute is changed to `inline-block`:

```html
<button class="trigger focusable visually-hidden" type="button" aria-pressed="false" 
  style="display: inline-block;">Open configuration options</button>
```

joeparsons commented on themes/custom/az_barrio/css/style.css:
```diff
@@ -718,3 +718,10 @@ fieldset.collapsed {
+/* Add ability to hide contextual links:
+ * See https://github.com/az-digital/az_quickstart/pull/2228
+ */
+ .hide-contextual-links .contextual button {
```
Whitespace nitpick:
```suggestion
.hide-contextual-links .contextual button {
```
